modules/table/src/main/java/org/apache/ignite/internal/table/distributed/SortedIndexLocker.java

```diff
@@ -100,7 +100,12 @@ public CompletableFuture<IndexRow> locksForScan(UUID txId, Cursor<IndexRow> curs
                 return null;
             }
 
-            return peekCursor.next();
+            IndexRow nextRow = peekCursor.next();
+
+            assert rowIdMatches(indexRow, nextRow) :
```

Comment:
I have no an answer. I think it will be possible, but how we will work with this...
This means the same code in a transaction can lead to different result.